scikeras/wrappers.py

```diff
@@ -1102,8 +928,8 @@ def predict_proba(self, X):
         # call the Keras model's predict
         outputs = self.model_.predict(X, **predict_args)
 
-        # join list of outputs into single output array
-        y = self.postprocess_y(outputs, return_proba=True)
+        # post process y
+        y = self.target_encoder_.inverse_transform(outputs, return_proba=True)
```

Comment:
The fact that `KerasClassifier.target_encoder_.inverse_transform` must accept a keyword arg `return_proba=True` will need documentation.

Reply:
Agreed. My attempt to document this (other than docstrings, which I still need to work on) was creating all of those base transformers so that each wrapper has a defined signature for each one of the transformer functions.

Reply:
Ok I reduced the classes to just the ones discussed plus `KerasFunctionTransformer` which is just `FunctionTransformer` but implementing `get_metadata`. This new class replaces several of the old ones.

And yes, `return_proba` is specific to `ClassifierLabelEncoder` / `KerasClassifier`. I'm open to ideas if you think there's a better way to do it than having that argument.

Reply:
> Ok I reduced the classes to just the ones discussed

Thanks!

> KerasFunctionTransformer which is just FunctionTransformer but implementing get_metadata

I'd remove `KerasFunctionTransformer`. I think it would be more convenient to users to use arbitrary Scikit-learn transformers out-of-the-box rather than create a new class, instantiate it before using it.

I'm thinking of the code below alongside the documentation "metadata will be collected from `get_metadata` if the transformer implements that method."

``` python
class BaseWrapper(...):
    def _fit(self, X, y):
        ...
        meta = transformer.get_metadata() if hasattr(transformer, "get_metadata") else {}
```

Reply:
Sure, we can do a `hasattr` instead.

By the way, I went with the `get_meta` because that method already exists for `BaseWrapper`.

Reply:
Implemented the `metadata will be collected from get_metadata if the transformer implements that method.` in the docstrings as well as removing the class and using `hasattr` instead. Let's resolve this thread?

Reply:
Resolved